Generate accessors for `settings.gradle.kts` (#27999)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

This fixes https://github.com/gradle/gradle/issues/12259.

> #### Accessors for `Settings` extensions

Reviewer: Can we make it known that its Kotlin specific in the title instead? I don't know if my suggestion makes sense but something like this:
```suggestion
#### Accessors for `Settings` extensions in Kotlin DSL
```

Reviewer 2: It wouldn't be a mistake, although `accessors` are Kotlin Dsl stuff essentially

Reviewer: @6hundreds, it's not a great suggestion, maybe something like:
```suggestion
#### Accessors for `Settings` extensions in Kotlin build files
```

> Previously, extensions registered in `Plugin<Settings>` weren't available in `settings.gradle.kts`. Access from Groovy work as expected.  

Reviewer: ```suggestion
Previously, extensions registered in `Plugin<Settings>` weren't available in `settings.gradle.kts`.
```

> Now, type-safe accessors for these extensions are being generating.

Reviewer: ```suggestion
Now, type-safe accessors for these extensions are generated.
```
